expcore/gui/prototype.lua

```diff
@@ -348,5 +350,11 @@ Gui._prototype_element.on_text_changed = event_handler_factory(defines.events.on
 -- @tparam function handler the event handler which will be called
 Gui._prototype_element.on_value_changed = event_handler_factory(defines.events.on_gui_value_changed)
 
+--- Custom element events.
+-- @section customEvents
+
+-- Triggered when a user changed the visibility of a left flow element by clicking a button
+Gui.events.on_visibility_changed_by_click = 'on_visibility_changed_by_click'
```

Comment:
Custom event section is ok for now depends how many events get added in the gui core. Non blocking.